packages/driver/src/cy/commands/actions/type.js

```diff
@@ -112,12 +112,13 @@ module.exports = function (Commands, Cypress, cy, state, config) {
       const isMonth = $dom.isType(options.$el, 'month')
       const isWeek = $dom.isType(options.$el, 'week')
       const hasTabIndex = $dom.isSelector(options.$el, '[tabindex]')
+      const isCurrentlyFocused = $dom.isFocused($dom.getElements(options.$el))
```

Comment:
It might make more sense to squash these into a generic `isFocusable`.

IMO `type()` shouldn't be chainable and just apply to the focused element but that be a breaking change that requires a lot of manual `focus()` calls and you could always restrict this manually in your codebase.